modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+  <http:listener path="account/{accountId}/main-contact" allowedMethods="GET" config-ref="HTTP_Listener_config"/>
+  <!-- fetch main contact for accountId -->
+</flow>
+
+<flow name="main-contact-general">
+  <http:listener path="account/{accountId}/main-contact" config-ref="HTTP_Listener_config"/>
+  <!-- not supported message -->
+</flow>
+----
 
-For example, a request to `+http://localhost:8081/account/mulesoft/main-contact+` maps the URI parameter to the Mule Message that holds the key `accountId` = `mulesoft`:
+When multiple listeners are defined, requests are routed to the first match in terms of their method, so default listeners should always be defined last.
 
-== Connector Mapping of Received Data
+=== From HTTP Request to Mule Message
 
-The data received by the connector during the Listener operation is always mapped as the payload of the propagated Mule Message.
+The metadata from the HTTP request line, including the method, request path, query and URI params as well as the headers are all available as attributes within a triggered flow.
+On the other hand, the body and the `Content-Type` header are used to set up the payload and its type. This allows other components to inspect the payload type: DataWeave for example, won't need any input information to work with an HTTP payload.
+Additionally, if an `X-Correlation-ID` or `MULE_CORRELATION_ID` (for compatibility) header is present, it will be used to set the message's correlation ID for traceability.
 
-=== Example of Receiving a Form
+Consider the following HTTP request:
 
-The following example shows the mapping by the HTTP Listener operation when an HTML form is received.
+[source,text,linenums]
+----
+POST api/v1/account/salesforce/main-contact?overwrite=true&notify=jane.doe&notify=admin HTTP/1.1
+Host: localhost:8081
+Content-Type: application/json
+Content-Length: 166
+X-Correlation-ID: 9cf32672-4f0b-4e8b-b988-40c13aae85b4
+
+{
+  "name": "John",
+  "surname": "Doe",
+  "role": "Senior Vice President",
+  "organization": "Marketing",
+  "phone": 701222369,
+  "email": "john.doe@salesforce.com"
+}
+----
 
-The body of a request is the following HTML form:
+The message's correlation ID will be `9cf32672-4f0b-4e8b-b988-40c13aae85b4` and its payload will be a JSON of:
+
+[source,text,linenums]
+----
+{
+  "name": "John",
+  "surname": "Doe",
+  "role": "Senior Vice President",
+  "organization": "Marketing",
+  "phone": 701222369,
+  "email": "john.doe@salesforce.com"
+}
+----
+
+An expression like `#[payload.name ++ ' ' ++ payload.surname]` will return `John Doe` since DataWeave will correctly interpret the JSON data.
+
+The attributes of the message will include:
+
+* method: `POST`
+* listenerPath: `api/v1/account/{accountId}/main-contact`
+* requestPath: `api/v1/account/salesforce/main-contact`
+* relativePath: `account/salesforce/main-contact`
+* queryParams: a multi map with entries `overwrite => true`, `notify => jane.doe` and `notify => admin`
```

Comment:
this should link to some reference of what a Multimap is and how to use it. The term will not automatically resonate with most users.

Also, that link should contain examples of how to access it. E.g: 

1) `vars.multimap.bleh` returns the first element
2)`vars.multimap.bleh[1]` returns the second element 
3) `vars.multimap.*bleh` returns all
4) `sizeOf(vars.multimap)` well.. you get the picture

Reply:
I added a note on this but will create a jira to add a full doc on it.